### PR TITLE
Agent strategy - check if caught in the last turn

### PR DESCRIPTION
**Pull request by kubitz**
# Summary

Resets LastTimeSinceCaught every time we are caught

## Additional Information

I reset the variable based on the SanctionScore, if it has increased, I assume we have been caught in the previous turn. (@Neelesh99 I believe that is a correct assumption right?)

## Test Plan

- ran go test. No issue detected

### CODE REVIEW

File: internal/clients/team3/clientgeneral.go [hunk 1]
```diff
@@ -44,6 +44,10 @@ func NewClient(clientID shared.ClientID) baseclient.Client {
 func (c *client) StartOfTurn() {
 	// c.Logf("Start of turn!")
 	// TODO add any functions and vairable changes here
+	var caught = c.checkIfCaught()
```
Comment (tharusha-a) [inline internal/clients/team3/clientgeneral.go]: could shorten to `if(c.checkIfCaught()) {c.timeSinceCaught = 0}`

File: internal/clients/team3/client.go [hunk 2]
```diff
@@ -62,6 +62,8 @@ type client struct {
 	params islandParams
 	// iigoInfo caches information regarding iigo in the current turn
 	iigoInfo iigoCommunicationInfo
+	// last sanction score cache to determine wheter or not we have been caugth in the last turn
+	last_sanctions roles.IIGOSanctionScore
```
Comment (tharusha-a) [inline internal/clients/team3/client.go:72]: This value isn't set anywhere. You should set this in `resetIIGOInfo` above the line that resets the sanctions struct (which I'm about to add whoops)

**Review by tharusha-a — CHANGES_REQUESTED** [review]
Added comments,  last_sanctions needs setting

File: internal/clients/team3/clientgeneral.go [hunk 3]
```diff
@@ -155,6 +159,15 @@ func (c *client) shouldICheat() bool {
...
 
+// checkIfCaught, checks if the island has been caught during the last turn
+// If it has been caught, it returns True, otherwise False.
+func (c *client) checkIfCaught() bool {
+	if c.iigoInfo.sanctions.ourSanction > c.last_sanctions {
+		return true
+	}
+	return false
+}
+
```
Comment (ramon15749) [inline internal/clients/team3/clientgeneral.go]: Nit: you could just do

`return c.iigoInfo.sanctions.ourSanction > c.last_sanctions`

**Pull request merged**